include/pcmki/pcmki_sched_utils.h

```diff
@@ -43,8 +43,8 @@ GList *sort_nodes_by_weight(GList *nodes, pe_node_t *active_node,
                             pe_working_set_t *data_set);
 
 extern gboolean can_run_resources(const pe_node_t * node);
-extern gboolean native_assign_node(pe_resource_t *rsc, pe_node_t *chosen,
-                                   gboolean force);
+extern gboolean native_assign_node(pe_resource_t *rsc, pe_working_set_t *data_set,
+                                   pe_node_t *chosen, gboolean force);
```

Comment:
Actually you don't need this -- you can use rsc->cluster

Reply:
Fixed.  Nice to remove the rest of the crud from that patch.  It's much clearer now.